src/Microsoft.IdentityModel.Validators/AadTokenValidationParametersExtension.cs

```diff
@@ -109,6 +138,43 @@ internal static bool ValidateIssuerSigningKey(SecurityKey securityKey, SecurityT
             return true;
         }
 
+        /// <summary>
+        /// Validates the cloud instance name of the signing key in a property bag.
+        /// </summary>
+        /// <param name="securityKey">The <see cref="SecurityKey"/> that signed the <see cref="SecurityToken"/>.</param>
+        /// <param name="configuration">The <see cref="BaseConfiguration"/> provided.</param>
+        /// <param name="cloudInstanceName">The cloud instance name to validate against.</param>
+        internal static void ValidateSigningKeyCloudInstanceName(SecurityKey securityKey, BaseConfiguration configuration, string cloudInstanceName)
+        {
+            if (securityKey == null)
+                return;
+
+            if (configuration is not OpenIdConnectConfiguration openIdConnectConfiguration)
+                return;
+
+            JsonWebKey matchedKeyFromConfig = openIdConnectConfiguration.JsonWebKeySet?.Keys.FirstOrDefault(x => x != null && x.Kid == securityKey.KeyId);
+            if (matchedKeyFromConfig != null && matchedKeyFromConfig.AdditionalData.TryGetValue(OpenIdProviderMetadataNames.CloudInstanceName, out object value))
+            {
+                string signingKeyCloudInstance = value as string;
+                if (string.IsNullOrWhiteSpace(signingKeyCloudInstance))
+                    return;
+
+                // Store the cloud instance name in the security key's property bag.
+                securityKey.PropertyBag[OpenIdProviderMetadataNames.CloudInstanceName] = signingKeyCloudInstance;
```

Comment:
Let's think about thread safety of this collection.

Reply:
thanks, this code was reverted